src/connection_deduplicator.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// http://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+use super::connections::Connection;
+use err_derive::Error;
+use std::{
+    collections::{hash_map::Entry, HashMap},
+    net::SocketAddr,
+};
+use tokio::sync::{broadcast, Mutex};
+
+type Result<T = Connection, E = Error> = std::result::Result<T, E>;
```

Comment:
Is this a common practice? I find it weird as when reading the code i cannot see what a function returns...? shouldn't be better to keep `T` to be set by insstantiator ?

Reply:
This is just to reduce typing. It is not exposed outside of this module. 

> Is this a common practice

Well, I can think of at least one precedent: https://doc.rust-lang.org/std/fmt/type.Result.html